libpromises/evalfunction.c

```diff
@@ -2775,6 +2847,22 @@ static FnCallResult FnCallMapData(EvalContext *ctx, ARG_UNUSED const Policy *pol
         return FnFailure();
     }
 
+    if (mapdatamode && jqmode)
+    {
+        // TODO: make this customizable
```

Comment:
Yeah, it should be enough for now. I just changed `cf_popen_full_duplex()` in a backwards-compatible way.